### DISCUSSION TITLE
Stopping solver once condition is met

### DISCUSSION THREAD

**Discussion by francescabasini** [zwicker-group/py-pde#294] [General]
Dear Dr. Zwicker,

Thank you for the extremely useful package.

I wonder if there exists a way to have a tracker that checks for a condition on the system's state and stops the iterations while saving the latest result. 
For now I was able to use `DataTracker` to compute the quantity of interest (a float) at a fixed interval but the solver runs until t_range = t_max. 
I'd very much appreciate an input on this.

**Comment by david-zwicker** [comment]
Your best bet might be using a [`CallbackTracker`](https://py-pde.readthedocs.io/en/latest/packages/pde.trackers.trackers.html#pde.trackers.trackers.CallbackTracker), which calls a user-defined function. This function can of course store data wherever you want it to be, but it can also raise a `StopIteration` exception, which will abort the simulation cleanly.
